Revert "Centralize fallback sample rate with kFallbackSampleRate"

sukibaby opened the pull request:
Putting it back to the way it was, because this particular commit did introduce some regressions, most significantly in RageSoundReader_Chain

teejusb commented:
I actually think the original change is still very good because centralizing a constant throughout the codebase is definitely preferred. I think in this case we could just revert the RageSoundReader_Chain changes from

```cpp
m_iPreferredSampleRate = 44100;
```
to
```cpp
m_iPreferredSampleRate = kFallbackSampleRate;
```

And revert the conditional/preference things i.e. this stuff:

```cpp
m_iPreferredSampleRate = PREFSMAN->m_iSoundPreferredSampleRate;
if (m_iPreferredSampleRate == 0)
{
	m_iPreferredSampleRate = kFallbackSampleRate;
}
```

sukibaby commented:
> I actually think the original change is still very good because centralizing a constant throughout the codebase is definitely preferred.

When making the PR, I thought it would be ugly to have a hardcoded sample rate in RageSoundReader but use the hardcoded sample rate elsewhere,  but maybe that actually makes it more clear to the reader that value is intended to be separate from the constant?

sukibaby commented:
I agree having a constant is better, but isn't this code being done away with soon, after the introduction of a replacement audio library is complete?

i figured removing as many variables that set it apart from SM5.1 behavior would be best, hence the reverts to hard coded 44100 - but I have also opened #987 which is just the RageSoundReader change

teejusb commented:
> I agree having a constant is better, but isn't this code being done away with soon, after the introduction of a replacement audio library is complete?
> 
> i figured removing as many variables that set it apart from SM5.1 behavior would be best, hence the reverts to hard coded 44100 - but I have also opened #987 which is just the RageSoundReader change

Centralizing to a constant is better practice and definitely what I'd push for. It won't have a performance impact either so I'd still push for doing the right thing here instead of having magic numbers littered throughout. I can merge #987 instead  as well, but would still prefer to utilize kFallbackSampleRate instead of hardcoding.

sukibaby commented:
I will update this one and close #987 .

sukibaby commented:
Sorry this took so long.

sukibaby commented:
@teejusb  

 I can't rebase this

Not sure what's up with that so lmk if you need this remade

<img width="378" height="195" alt="image" src="https://github.com/user-attachments/assets/cda20cdd-e669-481e-abb2-bc26a6c40871" />

teejusb commented on src/global.h:
```diff
@@ -127,6 +127,10 @@ void CallEveryNFrames(int n, Func&& f, Args&&... args) {
 /* Don't include our own headers here, since they tend to change often. */
 
+// Global fallback sample rate for audio.
+// This is here to avoid creating circular dependencies within RageSound.
```
Wait I don't follow why this is being moved to global. Seems like a step backwards and also not what we had prior.

sukibaby replied:
I didn't want to include RageSound.h in RageSoundDriver.h, because that's the way it always was until it was added so RageSoundDriver.h could have access to kFallbackSampleRate.

I don't want to include RageSound.h there, so, to get around that, I could hardcode 44100 or 48000 into RageSoundDriver.h? It's just the fallback value for the virtual function, so we don't even expect to return that one anyway.

Just trying to avoid including RageSound.h in RageSoundDriver.h mainly.

sukibaby replied:
RageSoundDriver.h needs knowledge of kFallbackSampleRate, but traditionally RageSound.h was never included in the RageSoundDriver files, and I don't think it should be.

Will make a better implementation than this and push though.   (I also just remembered I think I already covered this in the automatic sample rate detection PR which i had forgot about).

sukibaby commented:
@teejusb please lmk if any changes are requested, this is the only pr i want to be sure gets in before 1.2.0

sukibaby committed:
Set default sample rate to 44100 Hz in Chain reader

Replaces usage of PREFSMAN for preferred sample rate with a hardcoded default of 44100 Hz in RageSoundReader_Chain. Adds clarifying comments for sample rate variables to improve code readability.

sukibaby committed:
Refactor GetSampleRate in RageSoundReader_Chain

Moved GetSampleRate implementation from header to source file and updated logic to return m_iPreferredSampleRate if m_iActualSampleRate is -1. This improves clarity and ensures correct sample rate is reported when actual rate is not set.

sukibaby commented:
Superceded by #1041
